File: Java/src/tema9programacionorientadaaobjetos/excepciones/ejercicio2/TestGato.java
package tema9programacionorientadaaobjetos.excepciones.ejercicio2;

/**
 * Programa que prueba la clase GatoSimple
 *
 * @author Fco Javier Frías Serrano
 * @version 1.0
 */
public class TestGato {

	public static void main(String[] args) {

		Gato garfield = new Gato("macho");
		Gato tom = new Gato("macho");
		Gato lisa = new Gato("hembra");

		try {
			garfield.apareaCon(tom);
//			garfield.apareaCon(lisa);
		} catch (ApareamientoImposibleException e) {
			System.err.println("Imposible aparear. Los gatos solo pueden aparearse entre distintos sexos.");

		}

	}

}

File: Python/poo/excepciones/ejercicio2/TestGato.py
'''
Programa que prueba la clase GatoSimple

@author: Francisco Javier Frías Serrano
'''
from poo.excepciones.ejercicio2.Gato import Gato

garfield = Gato("macho")
tom = Gato("macho")
lisa = Gato("hembra")

try:
    garfield.apareaCon(lisa)
    garfield.apareaCon(tom)
except:
    print()